### PR TITLE
set small off diagonal elements of Delta_infty to 0

### PR DESCRIPTION
**Pull request by jkarp314**
I added a parameter "off_diag_threshold" which is by default 1e-13. If the off diagonal elements of Delta_infty are less than this threshold, then they are set to 0.

### CODE REVIEW

File: c++/triqs_cthyb/parameters.hpp [hunk 1]
```diff
@@ -234,6 +234,10 @@ namespace triqs_cthyb {
 
     /// Read solve_parameters_t from hdf5
     friend void h5_read(h5::group h5group, std::string subgroup_name, solve_parameters_t &sp);
+
+    /// Threshold below which which off diagonal components of hloc are set to 0
+    double off_diag_threshold = 1.e-13;
+
```
Comment (Wentzell) [inline c++/triqs_cthyb/parameters.hpp]: I would opt to not drop off-diagonal elements by default, but instead default to zero here

```suggestion

    /// Threshold below which which off diagonal components of hloc are set to 0
    double off_diag_threshold = 0.0;
```

File: c++/triqs_cthyb/solver_core.cpp [hunk 2]
```diff
@@ -154,6 +154,16 @@ namespace triqs_cthyb {
...
+    // set off diagonal terms in Delta_infty to 0 if they are below off_diag_threshold
+    for (int b : range(gf_struct.size())){
+      for (int i : range(first_dim(Delta_infty_vec[b]))){
+        for (int j : range(second_dim(Delta_infty_vec[b]))){
+          if (i != j && abs(Delta_infty_vec[b](i,j)) < params.off_diag_threshold)
+            Delta_infty_vec[b](i,j) = 0;
+        }
+      }
+    }
+
```
Comment (Wentzell) [inline c++/triqs_cthyb/solver_core.cpp]: To be consistent with the `imag_threshold` behavior I suggest to only set the corresponding terms to zero in the h_loc operator, not in `Delta_infty_vec`
```suggestion
```

For this it should be sufficient to add an additional check in the h_loc initialization below line 182

```cpp
  if(n1 != n2 && abs(Delta_infty_vec[b](n1, n2)) < params.off_diag_threshold) e_ij = 0.0;
```

**Review by Wentzell — CHANGES_REQUESTED** [review]
Dear @jkarp314,

Thank you for this addition!  See my comments / change requests below

**Comment by Wentzell** [comment]
Can you also regenerate the Python bindings and commit the updated solver_core_desc.py file?
```
cd path_to_src/python/triqs_cthyb
c++2py -r solver_core_desc.py
```

**Comment by jkarp314** [comment]
@Wentzell I think I've dealt with all of your comments and suggestions. How does it look now?

**Pull request merged**